### PR TITLE
Feature/auth updates

### PR DESCRIPTION
**Pull request by edward93**
No more getting logged out when refreshing the page.
Created new httpClient to hide the implementation details and make transition to other package i.e. fetch api, much easier.

Other QOL changes

### CODE REVIEW

File: redux/store.ts [hunk 1]
```diff
@@ -38,19 +38,20 @@ const rootReducer = (state: RootState = defaultState, action: AppAction) => {
...
-                const stateWithAuth: RootState = {
-                    ...state,
-                    user: {
-                        ...state.user,
-                        authHeaders: action.payload.authHeaders,
-                    },
-                };
-                return combinedReducer(stateWithAuth, action);
-            } else return combinedReducer(state, action);
+            // if (
```
Comment (kwf37) [inline redux/store.ts:41]: Can delete since we're moving this stuff to local storage

File: services/auth.helper.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+export type AuthHeaders = {
+    client: string;
+    expiry: string;
+    uid: string;
+    'access-token': string;
+    'token-type': string;
+};
+
+/**
+ * Stores logged in user id
```
Comment (kwf37) [inline services/auth.helper.ts:14]: I think these comments should be a bit more detailed so they're more friendly to someone new to the project. I'm happy to take a look and try editing them as well.

File: services/httpClient.ts [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    AuthHeaders,
+    clearAuthInfo,
+    getAuthInfo,
+    storeAuthInfo,
+} from './auth.helper';
+
+const baseURL = process.env.NEXT_PUBLIC_BASE_URL;
+
+/**
+ * Makes a 'GET' request
```
Comment (kwf37) [inline services/httpClient.ts:18]: Should add notes about the auth side effects of this function.

File: services/auth.helper.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
+const storageKey = 'exp-trk';
```
Comment (kwf37) [inline services/auth.helper.ts:1]: Maybe add a Constants comment banner and Export Functions comment banner. We've been copying our comment banners to all our new files

File: services/httpClient.ts [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+//                             Imports
+// ===================================================================
+import axios, { AxiosResponse } from 'axios';
+import {
+    AuthHeaders,
+    clearAuthInfo,
+    getAuthInfo,
+    storeAuthInfo,
+} from './auth.helper';
+
```
Comment (kwf37) [inline services/httpClient.ts:14]: ditto on Constants and Functions comment banner.

**Review by kwf37 — CHANGES_REQUESTED** [review]
- I like the new env file for declaring constants
- Love the new HttpClient module! It's great that it handles the authentication stuff internally and makes it easier to swap request libraries.
- Love the removal of all the redundant authHeader stuff!
Overall looks great- I added some inline requests for changes.

File: services/httpClient.ts (hunk 5 above)
Comment (edward93) [reply services/httpClient.ts:14]: Done. I think :)

File: services/auth.helper.ts (hunk 4 above)
Comment (edward93) [reply services/auth.helper.ts:1]: done

File: services/httpClient.ts (hunk 3 above)
Comment (edward93) [reply services/httpClient.ts:18]: I added few more lines but I feel like I'm not quite covering everything. If you could tell me a bit more perhaps I could improve it

File: services/auth.helper.ts (hunk 2 above)
Comment (edward93) [reply services/auth.helper.ts:14]: Same as the bellow one

File: redux/store.ts (hunk 1 above)
Comment (edward93) [reply redux/store.ts:41]: done

File: pages/_app.tsx [hunk 6]
```diff
@@ -25,4 +26,6 @@ function MyApp({ Component, pageProps }: AppProps) {
     );
 }
 
-export default wrapper.withRedux(MyApp);
+export default dynamic(() => Promise.resolve(wrapper.withRedux(MyApp)), {
```
[the remark this answers is not in the document]
Comment (edward93) [reply pages/_app.tsx:29]: done

**Review by kwf37 — APPROVED** [review]
Looks good enough for now! I'm gonna post a minor issue to remind me about those eslint no-explicit any waivers cause I think we might wanna make those nicer later.

**Pull request merged**